### PR TITLE
Add project module

### PR DESCRIPTION
**Pull request by tylerauerbeck**
##### SUMMARY
Begins to address #71 by adding the ability to manage a DO project

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
digital_ocean_project 

##### ADDITIONAL INFORMATION
This adds the ability to manage a digital ocean project via this collection.

### CODE REVIEW

File: plugins/modules/digital_ocean_project.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+# This creates a project that results with a purpose of "Other: My Prod App"
+- community.digitalocean.digital_ocean_project:
+    name: "ProdProj"
+    state: "present"
+    description: "This is a prod app"
+    purpose: "My Prod App"
+    environment: "Production"
+
+# This removes a project
+- community.digitlocean.digital_ocean_project:
```
Comment (mamercad) [inline plugins/modules/digital_ocean_project.py:110]: Small typo in this example "digitlocean".

File: tests/integration/targets/digital_ocean_project/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      community.digitalocean.digital_ocean_project:
+        oauth_token: "{{ do_api_key }}"
+        state: absent
+        name: "my-test-project"
+      register: result
+      with_items:
+        - "my-test-project"
+        - "my-updated-test-project"
+
+    - name: Verify Droplet is absent
```
Comment (mamercad) [inline tests/integration/targets/digital_ocean_project/tasks/main.yml:21]: Fixup name to say "Project" not "Droplet".

File: tests/integration/targets/digital_ocean_project/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      community.digitalocean.digital_ocean_project:
+        oauth_token: "{{ do_api_key }}"
+        state: present
+        name: "my-test-project"
+        purpose: "IoT"
+        description: "This is a test project"
+        environment: "Development"
+      register: result
+
+    - name: Verify Droplet is present (from present)
```
Comment (mamercad) [inline tests/integration/targets/digital_ocean_project/tasks/main.yml:36]: "Project"

File: tests/integration/targets/digital_ocean_project/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+      community.digitalocean.digital_ocean_project:
+        oauth_token: "{{ do_api_key }}"
+        state: present
+        name: "my-test-project"
+        purpose: "IoT"
+        description: "This is a test project"
+        environment: "Development"
+      register: result
+
+    - name: Verify Droplet is present (from existing)
```
Comment (mamercad) [inline tests/integration/targets/digital_ocean_project/tasks/main.yml:55]: "Project"

**Review by mamercad — CHANGES_REQUESTED** [review]
Very nice, just small things to fix, thank you!

**Comment by tylerauerbeck** [comment]
Good catch! These should be fixed up now.

**Review by mamercad — APPROVED** [review]
LGTM, thanks much!

**Pull request merged**